presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -480,4 +500,141 @@ private static int findGroupEnd(PagesIndex pagesIndex, PagesHashStrategy pagesHa
         }
         return endPosition;
     }
+
+    public static class WindowInfo
+            implements Mergeable<WindowInfo>, OperatorInfo
+    {
+        private final List<DriverWindowInfo> windowInfos;
+
+        @JsonCreator
+        public WindowInfo(@JsonProperty("windowInfos") List<DriverWindowInfo> windowInfos)
+        {
+            this.windowInfos = ImmutableList.copyOf(windowInfos);
```

Comment:
`ImmutableList.copyOf` internally check for non-nullness, so it's not necessary to add extra `requireNonNull` here.